phoenix-core/src/it/java/org/apache/phoenix/end2end/index/GlobalIndexCheckerIT.java

```diff
@@ -167,6 +169,129 @@ public void testDelete() throws Exception {
         }
     }
 
+    @Test
+    public void testPhoenixRowTimestamp() throws Exception {
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            String dataTableName = generateUniqueName();
+            String indexTableName = generateUniqueName();
+            Timestamp initial = new Timestamp(EnvironmentEdgeManager.currentTimeMillis() - 1);
+            conn.createStatement().execute("create table " + dataTableName +
+                    " (id varchar(10) not null primary key, val1 varchar(10), val2 varchar(10), val3 varchar(10))" + tableDDLOptions);
+            conn.createStatement().execute("upsert into " + dataTableName + " values ('a', 'ab', 'abc', 'abcd')");
+            conn.commit();
+            Timestamp before = new Timestamp(EnvironmentEdgeManager.currentTimeMillis());
+            // Sleep 1ms to get a different row timestamps
+            Thread.sleep(1);
+            conn.createStatement().execute("upsert into " + dataTableName + " values ('b', 'bc', 'bcd', 'bcde')");
+            conn.commit();
+            Timestamp after = new Timestamp(EnvironmentEdgeManager.currentTimeMillis() + 1);
+            conn.createStatement().execute("CREATE INDEX " + indexTableName + " on " +
+                    dataTableName + " (val1, PHOENIX_ROW_TIMESTAMP()) " + "include (val2, val3) " + (async ? "ASYNC" : "")+ this.indexDDLOptions);
+            if (async) {
+                // Run the index MR job to rebuild the index and verify that index is built correctly
+                IndexToolIT.runIndexTool(true, false, null, dataTableName,
+                        indexTableName, null, 0, IndexTool.IndexVerifyType.AFTER);
+            }
+
+            String timeZoneID = Calendar.getInstance().getTimeZone().getID();
+            // Write a query to get the val2 = 'bc' with a time range query
+            String query = "SELECT  val1, val2, PHOENIX_ROW_TIMESTAMP() from " + dataTableName + " WHERE val1 = 'bc' AND " +
```

Comment:
Mind adding some /*+ NO_INDEX */ cases? (if they break disable them, but would be important to have)

Reply:
I attempted to add a case with /*+ NO_INDEX */  but it failed. it seems the PHOENIX_ROW_TIMESTAMP() does not work with MultiKeyValueComparisonFilter. Please note this is not a just  with the no index hint, it is a bug in general for data table with or without an index. I will file a bug on this.

Reply:
I noticed that my test for /*+ NO_INDEX */  was wrong and that's why it failed. I/*+ NO_INDEX */  worked for me.

Reply:
Nice!
I definitely found that SELECT COUNT(DISTINCT(phoenix_row_timestamp())) FROM T returns the wrong results. That is due to an incorrect plan being generated, though, not related to this. I filed a separate jira for that.